### PR TITLE
FFWEB-1849: `idType` attribtue for ff-similar-products

### PR DESCRIPTION
**Pull request by a-laurowski**

### CODE REVIEW

**Review by fmarangi — APPROVED** [review]

File: markdown/3.x/en/api/ff-similar-products.api.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ ___
 | Name | Description |
 | ---- | ----------- |
 | **record-id**&nbsp;(Number) (default: empty) | The record ID of the product, for which similar products are returned. This also works for multiple products. In this case the record IDs have to be separated with `,`. |
+| **id-type**&nbsp;(String) **Options**: &nbsp;"id", &nbsp;"productNumber" (default: undefined) | The type of identifier for which the corresponding value should be passed in the **record-id**. This attribute is mandatory when using FACT-Finder NG. |
```
Comment (omikron-fp) [inline markdown/3.x/en/api/ff-similar-products.api.md:7]: Honestly, I don't understand what this is trying to tell me.

Considering the FF API documentation
> Use this parameter to determine the type of id passed to the method.

Perhaps we can make something short and simple, too?

```suggestion
| **id-type**&nbsp;(String) **Options**: &nbsp;"id", &nbsp;"productNumber" (default: undefined) | This attribute determines the type of `record-id`. It is mandatory when using FACT-Finder NG. |
```

**Review by omikron-fp — CHANGES_REQUESTED** [review]

File: markdown/3.x/en/api/ff-similar-products.api.md (hunk 1 above)
Comment (a-laurowski) [reply markdown/3.x/en/api/ff-similar-products.api.md]: Yes i had difficulties to to put it somehow neatly, this version is much better. Changed in v4 too

**Comment by omikron-fp** [comment]
Shall we introduce the (required/conditionally required) pattern like in `ff-communication`?

**Comment by a-laurowski** [comment]
> Shall we introduce the (required/conditionally required) pattern like in `ff-communication`?

sure, we can. 
1. Do we update only v4? (like ff-communication)
2. Should we use"required" mark only with attributes which will not prevent element from working but will eventually break the execution and emit warning/error (like api,version, channel and rest from ff-communication), or include also these, which while not being set, will make element will not work at all (like record-id from this element)

**Comment by omikron-fp** [comment]
Only applying it to `4.x` should be fine for now.
I think it would be best to mark all attributes as _required_ that are needed for proper functioning. So here it would be `record-id` and `id-type` I guess. Not sure though whether `id-type` isn't conditionally required.

**Comment by a-laurowski** [comment]
> Only applying it to `4.x` should be fine for now.
> I think it would be best to mark all attributes as _required_ that are needed for proper functioning. So here it would be `record-id` and `id-type` I guess. Not sure though whether `id-type` isn't conditionally required.

ah yes, its mandatory only for NG. changed this and added record-id as required

**Review by omikron-fp — APPROVED** [review]

**Pull request merged**